packages/core/src/http-handler.ts

```diff
@@ -60,6 +67,23 @@ export class HttpHandler {
     return requestContext;
   }
 
+  protected _bindAuthenticationFn(context: Context): void {
+    context.bind('authenticate').toDynamicValue((): Authenticate => {
+      return async (request: ParsedRequest) => {
+        const meta: AuthenticationMetadata =
+          await context.get(BindingKeys.Context.CONTROLLER_METHOD_META);
+        if (!meta)
+          return;
+        const strategy = context.get(meta.strategy);
```

Comment:
IIUC, this requires two steps in order to get the strategy instance:
 1. resolve CONTROLLER_METHOD_META
 2. resolve `meta.strategy`

It makes me wonder whether we can rewrite this in a more elegant way, one that will not require us to call `ctx.get` that much. Perhaps `AUTHENTICATION_STRATEGY` should be bound by the same code that's binding `CONTROLLER_METHOD_META`, so that `authenticateFn` provider can get the authentication strategy injected directly?

The more I think about this code, the less I know how to make it cleaner. I guess my main concern is that the proposed implementation of `authenticationFn` is doing too much work at too many different levels of abstractions. Ideally, all dependencies of `authenticateFn` should be registered/bound somewhere else, so that the body of this function can become just a small lambda delegating the work to the passport-adapter function.

```ts
class AuthenticationFnProvider {
  constructor(@inject(AUTHENTICATION_PROVIDER) private impl: AuthenticateFn) {
  }

  value() {
    return async (request: ParsedRequest): UserProfile => {
      return await impl(request);
    }
  }
}
```